frame/staking/src/default_weights.rs

```diff
@@ -0,0 +1,249 @@
+
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Default weights of pallet-staking.
+//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 2.0.0-rc6
+
+#![allow(unused_parens)]
+#![allow(unused_imports)]
+
+use frame_support::weights::{Weight, constants::RocksDbWeight as DbWeight};
+
+pub struct WeightInfo;
+impl crate::WeightInfo for WeightInfo {
```

Comment:
IIRC we wanted to define the Substrate/Polkadot based weights as the default, `impl crate::WeightInfo for ()`

Reply:
Yes, for tests it is the most ergonomic

Reply:
I want to object to that that using `()` for a default seems a bit strange to me. My proposal is to have `()` implement something _dummy_ (which is sensible for its null-ness) and some struct implement it for _defaults_. The former can be std-only to ensure it is never used in reality. WDY-gusy-T?

Reply:
Well, the thought is that there are many existing substrate chains which are all taking from our "default" weights anyway, coming from our pallet.
For them, they have already added `WeightInfo = ()` which was introduced earlier, and as we migrate to weightinfo, we should do a non-breaking change to them, imo.

Reply:
Ambivalent and will change to be consistent with other pallets. But I would do it differently and not use `()`.